lib/ansible/modules/windows/win_partition.ps1

```diff
@@ -0,0 +1,317 @@
+        PartitionNumber = $partition.PartitionNumber
+    }
+
+    if ($hidden -NotIn ($null, $partition.IsHidden)) {
+        $parameters.Add("IsHidden", $hidden)
+    }
+
+    if ($read_only -NotIn ($null, $partition.IsReadOnly)) {
+        $parameters.Add("IsReadOnly", $read_only)
+    }
+
+    if ($active -NotIn ($null, $partition.IsActive)) {
+        $parameters.Add("IsActive", $active)
+    }
+
+    try {
+        Set-Partition @parameters
+    } catch {
+        Fail-Json -obj $result -message "Error changing state of partition: $($_.Exception.Message)"   
+    }
+}
+
+
+if ($ansible_partition) {
+    if ($state -eq "absent") {
+        try {
+            Remove-Partition -DiskNumber $ansible_partition.DiskNumber -PartitionNumber $ansible_partition.PartitionNumber -Confirm:$false -WhatIf:$check_mode
+        } catch {
+            Fail-Json -obj $result -message "There was an error removing the partition: $($_.Exception.Message)"
+        }
+        $result.changed = $true
+    }
+    else {
+
+        if ($null -ne $gpt_type -and $gpt_styles.$gpt_type -ne $partition.GptType) {
+            Fail-Json -obj $result -message "gpt_type is not a valid parameter for existing partitions"
+        }
+        if ($null -ne $mbr_type -and $mbr_styles.$mbr_type -ne $partition.MbrType) {
+            Fail-Json -obj $result -message "mbr_type is not a valid parameter for existing partitions"
+        }
+
+        if ($partition_size) {
+            try {
+                $max_supported_size = (Get-PartitionSupportedSize -DiskNumber $ansible_partition.DiskNumber -PartitionNumber $ansible_partition.PartitionNumber).SizeMax
+            } catch {
+                Fail-Json -obj $result -message "Unable to get maximum supported partition size: $($_.Exception.Message)"
+            }
+            if ($size_is_maximum) {
+                $ansible_partition_size = $max_supported_size
+            }
+            if ($ansible_partition_size -ne $ansible_partition.Size -and $ansible_partition_size -le $max_supported_size) {
+                if ($ansible_partition.IsReadOnly) {
+                    Fail-Json -obj $result -message "Unable to resize partition: Partition is read only"
+                } else {
+                    try {
+                        Resize-Partition -DiskNumber $ansible_partition.DiskNumber -PartitionNumber $ansible_partition.PartitionNumber -Size $ansible_partition_size -WhatIf:$check_mode
+                    } catch {
+                        Fail-Json -obj $result -message "Unable to change partition size: $($_.Exception.Message)"
+                    }                    
+                    $result.changed = $true
+                }
+            } elseif ($ansible_partition_size -gt $max_supported_size) {
+                Fail-Json -obj $result -message "Specified partition size exceeds size supported by partition"
+            }
+        }
+
+        if ($drive_letter -NotIn ("auto", $null, $ansible_partition.DriveLetter)) {
```

Comment:
What happens when `$drive_letter` is `$null`?

Reply:
It won't do anything.

If the user wants to remove the `drive_letter` (I assume that's what they're trying by keeping it blank), I wanted to propose a separate module.

From the OP:

> Removing drive letters from partitions isn't supported by this. I was leaning towards a win_access_path module for adding/removing mount points instead.

Reply:
Ah that make sense now, sorry I don't know what I was thinking when I posted the comment.